src/mesh.cpp

```diff
@@ -3525,9 +3586,9 @@ int LibMesh::get_bin(Position r) const
   }
 
   const auto& point_locator = pl_.at(thread_num());
+  const auto element = (*point_locator)(p);
 
-  const auto elem_ptr = (*point_locator)(p);
-  return elem_ptr ? get_bin_from_element(elem_ptr) : -1;
+  return element? get_bin_from_element(clustering_element_mapping_.at(element)):-1;
```

Comment:
Instead of always using this map, do a check in `get_bin_from_element` like they already do for adaptivity.

No changes are needed in this method -undo changes.

Reply:
At the beginning of `get_bin_from_element`:
```
   auto tally_elem = mesh_tally_amalgamation_valid_ ?
      clustering_element_mapping_.at(elem) : elem;
   // replace `elem` with `tally_elem` in following line
```